graph_snyk_cve_sync.py

```diff
@@ -100,55 +103,86 @@ def force_run_ingestion(self):
         """Return if ingestion mode is on."""
         return os.environ.get('SNYK_INGESTION_FORCE_RUN', 'false').lower() in ('1', 'yes', 'true')
 
+    def is_delta_mode_on(self):
+        """Return if the delta feed mode is on."""
+        return os.environ.get('SNYK_DELTA_FEED_MODE', 'false').lower() in ('1', 'yes', 'true')
+
 
 class SnykCveSync:
     """Snyk class to sync cves to graph."""
 
-    def __init__(self):
+    def __init__(self, data):
         """Init method for helper class."""
         self.helper = Helper()
         self.CVE_DATA = {}
-
-    def get_version_rules_npm(self, vuln_versions):
-        """Version rules for npm eco."""
-        rules = []
-        for version in vuln_versions:
-            rules.append(NpmSpec(version))
-        return rules
+        self.utc_now = datetime.utcnow()
+        if not data:
+            self.snyk_data = self.helper.read_data_from_s3(self.utc_now.strftime('%d-%m-%y'),
+                                                           "snyk-feed/")
+        else:
+            self.snyk_data = data
+
+    def parse_data_for_eco(self, eco):
+        """Return True/False depending on whether the ecosysytem data should be parsed or not."""
+        if self.helper.is_delta_mode_on():
+            return True
+        day = str(self.utc_now.weekday())
+        if ((day == "0" or day == "3") and eco == "java") \
+                or ((day == "1" or day == "4") and eco == "js") \
+                or ((day == "2" or day == "5") and eco == "python"):
+            return True
+        return False
 
     def get_version_rules(self, vuln_versions):
         """Version rules for pypi,maven eco."""
         rules = []
-        regex_op = "[0-9a-zA-Z.]+"
-        regex_vr = "[<>=]+"
+        regex_op = "[0-9a-zA-Z\\_\\.\\-]+"
+        regex_vr = "[<>=*]+"
         for version in vuln_versions:
+            version = version.replace(" ", "")
             if '||' in version:
```

Comment:
I dont think this if else is required. If there nothing to split by, it will return the first element.

Reply:
let me check on it

Reply:
done